### PR TITLE
Release v1.15.4

### PR DESCRIPTION
**Pull request by andrey-kashcheev**
Updated CHANGELOG and CMakeLists files for the release v1.15.4

Relates-To: OLPEDGE-2797

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,16 @@
+## v1.15.4 (13/03/2023)
+
+**Common**
+* Changed the log level in case of errors during cache open from `FATAL` to `ERROR`.
+* Changed network layer implementation based on cURL to handle certificates lookup.
+* Improved documentation.
+* Fixed `olp::utils::Dir::Size` to filter only files on Windows.
```
Comment (nplakhtii) [inline CHANGELOG.md:7]: Improved `olp::utils::Dir::Size` to filter only files on Windows.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,16 @@
...
+* Added limit to the number of attempts for cache compaction.
```
Comment (nplakhtii) [inline CHANGELOG.md:8]: Added a limit of attempts for cache compaction.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,16 @@
...
+**Common**
+* Changed the log level in case of errors during cache open from `FATAL` to `ERROR`.
+* Changed network layer implementation based on cURL to handle certificates lookup.
+* Improved documentation.
+* Fixed `olp::utils::Dir::Size` to filter only files on Windows.
+* Added limit to the number of attempts for cache compaction. 
+* Added `libssl` to the list of network libraries.
+
+**olp-cpp-sdk-dataservice-read**
+* Changed the log level of a few repetitive and unimportant log messages from `INFO` to `DEBUG`.
```
Comment (nplakhtii) [inline CHANGELOG.md:12]: Changed the log level of repetitive log messages from `INFO` to `DEBUG`.

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,16 @@
+## v1.15.4 (13/03/2023)
+
+**Common**
+* Changed the log level in case of errors during cache open from `FATAL` to `ERROR`.
```
Comment (nplakhtii) [inline CHANGELOG.md:4]: Changed the log level from `FATAL` to `ERROR` in case of errors during opening cache.

Comment (nplakhtii) [reply CHANGELOG.md]: Changed severity of a log level from FATAL to ERROR in case of errors that appear during opening cache.

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,3 +1,16 @@
...
+**Common**
+* Changed the log level from `FATAL` to `ERROR` in case of errors during opening cache.
+* Changed network layer implementation based on cURL to handle certificates lookup.
+* Improved documentation.
+* Improved `olp::utils::Dir::Size` to filter only files on Windows.
+* Added a limit of attempts for cache compaction.
+* Added `libssl` to the list of network libraries.
+
+**olp-cpp-sdk-dataservice-read**
+* Changed the log level of a repetitive log message from `INFO` to `DEBUG`.
```
Comment (nplakhtii) [inline CHANGELOG.md:12]: Changed the log level from `INFO` to `DEBUG` in case of repetitive log messages.

Comment (nplakhtii) [reply CHANGELOG.md:12]: Changed the log level of repetitive log messages from `INFO` to `DEBUG`.

**Review by mykhailo-kuchma — APPROVED** [review]

**Pull request merged**